pagopa/eng-lollipop-consumer-java-sdk
# [SLS-19] Rest client implementation for Identity Service

#### List of Changes
- Generated rest client using openAPI generator
- Defined methods that retrieves the idp certification data based on the given entityId and assertion issue instant
- Written tests to validate IdpCertSimpleClient class methods
- Removed unused classes from assertion-rest-client-native & identity-service-rest-client-native
- Defined two configuration classes for the respective ApiClient of IdpCertSimpleClient & AssertionSimpleClient

#### Motivation and Context
[Issue SLS-19 Rest client implementation for Identity Service](https://pagopa.atlassian.net/browse/SLS-19?atlOrigin=eyJpIjoiZWVlNmQ5YjkzY2FjNGU3NjhiYTYwZmQ4MDZkNTdhZDciLCJwIjoiaiJ9) 

#### How Has This Been Tested?
We've written tests for the following events:
- Retrieving successfully a SPID certification
- Retrieving successfully a CIE certification
- Verify that the correct error is shown when given a non-existing entityID
- Verify that the correct error is shown when given a wrong instant for SPID certifications
- Verify that the correct error is shown when given a wrong instant for CIE certifications

#### Types of changes
[X] Chore (nothing changes by a user perspective)
[X] Bug fix (non-breaking change which fixes an issue)
[  ] New feature (non-breaking change which adds functionality)
[  ] Breaking change (fix or feature that would cause existing functionality to not work as expected)

[SLS-19]: https://pagopa.atlassian.net/browse/SLS-19?atlOrigin=eyJpIjoiNWRkNTljNzYxNjVmNDY3MDlhMDU5Y2ZhYzA5YTRkZjUiLCJwIjoiZ2l0aHViLWNvbS1KU1cifQ
